src/wallet/rpcwallet.cpp

```diff
@@ -1491,23 +1491,10 @@ UniValue listtransactions(const JSONRPCRequest& request)
     if ((nFrom + nCount) > (int)ret.size())
         nCount = ret.size() - nFrom;
 
-    std::vector<UniValue> arrTmp = ret.getValues();
-
-    std::vector<UniValue>::iterator first = arrTmp.begin();
-    std::advance(first, nFrom);
-    std::vector<UniValue>::iterator last = arrTmp.begin();
-    std::advance(last, nFrom+nCount);
-
-    if (last != arrTmp.end()) arrTmp.erase(last, arrTmp.end());
-    if (first != arrTmp.begin()) arrTmp.erase(arrTmp.begin(), first);
-
-    std::reverse(arrTmp.begin(), arrTmp.end()); // Return oldest to newest
-
-    ret.clear();
-    ret.setArray();
-    ret.push_backV(arrTmp);
-
-    return ret;
+    const std::vector<UniValue>& txs = ret.getValues();
+    UniValue result{UniValue::VARR};
+    result.push_backV({ txs.rend() - nFrom - nCount, txs.rend() - nFrom });
```

Comment:
I didn't know what rend() did, so this seemed more explicit and has a simpler push_backV
```diff
+ const std::vector<UniValue, std::allocator<UniValue> >::const_reverse_iterator txs_end = txs.rend() - nFrom;
  UniValue result{UniValue::VARR};
- result.push_backV({ txs.rend() - nFrom - nCount, txs.rend() - nFrom });
+ result.push_backV({txs_end - nCount, txs_end});
```

Reply:
I'll leave the suggestion open for others to weight in.

Reply:
> I'll leave the suggestion open for others to weight in.

Just one opinion, but I prefer the current version without the extra variable

Reply:
FWIW kudos on @promag's response to leave it open for others to weigh in -- I noted it as a good example to follow.